drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1580 @@
+      return v
+    end
+  end
+  device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
+  return res
+end
+
+local function component_to_endpoint(device, component_name)
+  return find_default_endpoint(device, clusters.DoorLock.ID)
+end
+
+local function device_init(driver, device)
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  device:subscribe()
+ end
+
+local function device_added(driver, device)
+  device:emit_event(capabilities.lockAlarm.alarm.clear({state_change = true}))
+end
+
+local function do_configure(driver, device)
+  local user_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.USER})
+  local pin_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.PIN_CREDENTIAL})
+  local week_schedule_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.WEEK_DAY_ACCESS_SCHEDULES})
+  local year_schedule_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.YEAR_DAY_ACCESS_SCHEDULES})
+
+  local profile_name = "lock"
+  if #user_eps > 0 then
+    profile_name = profile_name .. "-user"
+    if #pin_eps > 0 then
+      profile_name = profile_name .. "-pin"
+    end
+    if #week_schedule_eps + #year_schedule_eps > 0 then
+      profile_name = profile_name .. "-schedule"
+    end
+  else
+    profile_name = "base-lock"
+  end
+  device.log.info(string.format("Updating device profile to %s.", profile_name))
+  device:try_update_metadata({profile = profile_name})
+end
+
+local function info_changed(driver, device, event, args)
+  for cap_id, attributes in pairs(subscribed_attributes) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, attr in ipairs(attributes) do
+        device:add_subscribed_attribute(attr)
+      end
+    end
+  end
+  for cap_id, events in pairs(subscribed_events) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, e in ipairs(events) do
+        device:add_subscribed_event(e)
+      end
+    end
+  end
+  device:subscribe()
+end
+
+-- Matter Handler
+----------------
+-- Lock State --
+----------------
+local function lock_state_handler(driver, device, ib, response)
+  local LockState = DoorLock.attributes.LockState
+  local attr = capabilities.lock.lock
+  local LOCK_STATE = {
+    [LockState.NOT_FULLY_LOCKED] = attr.not_fully_locked(),
+    [LockState.LOCKED] = attr.locked(),
+    [LockState.UNLOCKED] = attr.unlocked(),
+  }
+
+  -- The lock state is usually updated in lock_state_handler and lock_op_event_handler, respectively.
+  -- In this case, two events occur. To prevent this, when both functions are called,
+  -- it send the event after 1 second so that no event occurs in the lock_state_handler.
+  device.thread:call_with_delay(1, function ()
+    if ib.data.value ~= nil then
+      device:emit_event(LOCK_STATE[ib.data.value])
+    else
+      device:emit_event(attr.unknown())
```

Comment:
If `unknown` is going to be used here then it also needs to be added to `enabledValues` in the profiles. Otherwise it won't be translated correctly when that event is emitted.

Reply:
Okay I'll check it and modify.